### PR TITLE
Update LED information with Status LEDs

### PR DESCRIPTION
**Pull request by hamishwillee**
Attempts to capture info from https://github.com/PX4/Firmware/issues/10034#issuecomment-440787304

This is user facing, so does not include the pinouts. Questions inline.

### CODE REVIEW

File: en/flying/led_meanings.md [hunk 1]
```diff
@@ -41,3 +48,27 @@ this flight.
 Attach your autopilot to a Ground Control Station to verify what the problem is.
 If you have completed the setup process and autopilot still appears as red and flashing, there may be another error.
 
+## Status LED (FMUv5 and later) {#status_led}
+
+Three *Status LEDs* were added to the *Pixhawk Series* in Pixhawk 4 ([FMUv5](../flight_controller/pixhawk_series.md#fmu-versions)) to provide status for the PX4IO and FMU SoC. 
+They indicate power, bootloader mode and activity, and errors.
+
+The labeling and purpose of the LEDs on the PX4 reference board (HolyBro Pixhawk 4) for bootloading and when in normal use is given below:
```
Comment (hamishwillee) [inline en/flying/led_meanings.md]: Are the labels likely to be the same for any board? I assume not, so have specified HolyBro.

File: en/flying/led_meanings.md [hunk 2]
```diff
@@ -41,3 +48,27 @@ this flight.
...
+
+Color | Label | Bootloader usage | FMU usage 
+--- | --- | --- | ---
+Blue | ACT (Activity) | Flutters when the bootloader is receiving data | Indication of ARM
+Red/Amber | B/E (In Bootloader / Error) | Flutters when in the bootloader | Indication of an ERROR
+Green |PWR (Power) | Not used by bootloader | Indication of ARM state
+
+
+The meaning of the LEDs is as below (where "x" means "any state")
```
[the remark this answers is not in the document]
Comment (davids5) [reply en/flying/led_meanings.md]: There are 2 CPU on the board one is the PX4IO the other is the FMU. Both have bootloaders.
both act the same. 

One Caveat: There is a bug in the FMUv5 Bootloader deployed that flutteres the green LED not the Amber - but that will be fixed. So document the expected correct functionality.

File: en/flight_controller/pixhawk_series.md [hunk 3]
```diff
@@ -89,4 +89,7 @@ Product names/brands can also be trademarked. Trademarked names may not be used
 
 ### LEDs
 
-All *Pixhawk-series* flight controllers have an RGB LED to indicate the current status of the vehicle. To interpret the LEDs see: [LED Meanings](../flying/led_meanings.md).
+All *Pixhawk-series* flight controllers have an superbright RGB *UI LED* to indicate the current *readiness to fly* status of the vehicle.
```
Comment (davids5) [inline en/flight_controller/pixhawk_series.md]: > All *Pixhawk-series* flight controllers have an superbright RGB *UI LED* to indicate the current *readiness to fly* status of the vehicle. 

All *Pixhawk-series* flight controllers support user facing RGB *UI LED* to indicate the current *readiness to fly* status of the vehicle. This is typically a superbright I2C peripheral. 
  
   
Some are not on board. I..E are mounted on the GPS or a separate box. FmuV4 does not have one on board.

File: en/flying/led_meanings.md [hunk 4]
```diff
@@ -1,5 +1,12 @@
 # LED Meanings (Pixhawk Series)
 
+[Pixhawk-series flight controllers](../flight_controller/pixhawk_series.md) use LEDs to indicate the current status of the vehicle.
+- The [UI LED](#ui_led) has been present since Pixhawk 1 (FMUv2) and is used to provide user-facing status information related to *readiness for flight*.
+- The [Status LEDs](#status_led) were added in Pixhawk 4 (FMUv5), to provide status for the PX4IO and FMU SoC.
```
Comment (davids5) [inline en/flying/led_meanings.md]: not added. See Pictures.

File: en/flying/led_meanings.md [hunk 5]
```diff
@@ -41,3 +48,27 @@ this flight.
 Attach your autopilot to a Ground Control Station to verify what the problem is.
 If you have completed the setup process and autopilot still appears as red and flashing, there may be another error.
 
+## Status LED (FMUv5 and later) {#status_led}
+
+Three *Status LEDs* were added to the *Pixhawk Series* in Pixhawk 4 ([FMUv5](../flight_controller/pixhawk_series.md#fmu-versions)) to provide status for the PX4IO and FMU SoC.
```
Comment (davids5) [inline en/flying/led_meanings.md]: "

File: en/flying/led_meanings.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/flying/led_meanings.md]: @davids5 Thanks for all these clarifications

Got it for most of this. But here I see 5 LEDs, not three: 
image

Given you say above "Both have bootloaders. both act the same." are you actually saying that the FMU cpu has its set and PX4IO has its set? That's 6 in total, so wondering at the discrepancy.

In addition, how do Bootloader and FMU usage "relate" below - if both FMU and PX4IO have a bootloader then this is confusing.
image

File: en/flying/led_meanings.md [hunk 6]
```diff
@@ -41,3 +48,27 @@ this flight.
...
 If you have completed the setup process and autopilot still appears as red and flashing, there may be another error.
 
+## Status LED (FMUv5 and later) {#status_led}
+
+Three *Status LEDs* were added to the *Pixhawk Series* in Pixhawk 4 ([FMUv5](../flight_controller/pixhawk_series.md#fmu-versions)) to provide status for the PX4IO and FMU SoC. 
+They indicate power, bootloader mode and activity, and errors.
+
+The labeling and purpose of the LEDs on the PX4 reference board (HolyBro Pixhawk 4) for bootloading and when in normal use is given below:
+
+Color | Label | Bootloader usage | FMU usage
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/flying/led_meanings.md]: I asked for further clarification in other comment. But from above, perhaps the labels are wrong - ie "Bootloader usage" should be labeled "PX4IO usage"?

File: en/flying/led_meanings.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (davids5) [reply en/flying/led_meanings.md]: @hamishwillee 
Sorry fo the ambiguity. This is the classic issue of "I am too close to the solution" 

From power on all the PX4 CPUs first runs the bootloader (BL) then if present the application (APP). 

Bootloader usage is referring to how the bootloader uses the LED to indicate its condition. 
The FMU usage is referring to the application's usage of the LED. 

We have a common code in the BL and APP and consistent usage of the LEDs if those LEDs exist on a given CPU and board.  So the set of 3 LEDs make up the status annunciation. The super set  has to be documented and if a board has a subset that subset will be consistent with the usage.

File: en/flying/led_meanings.md [hunk 7]
```diff
@@ -41,3 +48,27 @@ this flight.
...
+## Status LED (FMUv5 and later) {#status_led}
+
+Three *Status LEDs* were added to the *Pixhawk Series* in Pixhawk 4 ([FMUv5](../flight_controller/pixhawk_series.md#fmu-versions)) to provide status for the PX4IO and FMU SoC. 
+They indicate power, bootloader mode and activity, and errors.
+
+The labeling and purpose of the LEDs on the PX4 reference board (HolyBro Pixhawk 4) for bootloading and when in normal use is given below:
+
+Color | Label | Bootloader usage | FMU usage 
+--- | --- | --- | ---
+Blue | ACT (Activity) | Flutters when the bootloader is receiving data | Indication of ARM
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/flying/led_meanings.md]: I think I get it now.

File: en/flying/led_meanings.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (hamishwillee) [reply en/flying/led_meanings.md]: Thanks @davids5 . I have updated based on my current understanding and added an image. I am fairly confident this is correct and easy understand now. Please confirm is OK.

**Comment by hamishwillee** [comment]
Thanks very much. Merged.

**Pull request merged**